### DISCUSSION TITLE
DataGrid Header and Bootstrap 5 Dark Mode

### DISCUSSION THREAD

**Discussion by Alexander-JM-0224** [Megabit/Blazorise#5711] [Q&A] [answered]
Hello,

I am currently working on the UI of the web application of my employer, and i am supposed to implement a switchable Dark Theme.

Since we use Bootstrap 5 (5.3.3) i have used the JS script in the Bootstrap docs to make switching our Bootstrap theme possible and this works wich is very nice.
There is no other styling in our application (Blazor WASM .NET8) for the DataGrids, but the background colors do not all switch correctly and the inspector seems to indicate there is background-color coming from the Blazorise CSS.
Important note, i have already tried to give the DataGrids their own CSS class just for the background color. Even with !important it does nothing for me. 
When i switch the color off in the inspector everything is perfect.

Problem DataGrid Header
![Header-pict](https://github.com/user-attachments/assets/239bf259-1659-4219-aeda-d0ab662c86f8)
Inspector
![Inspector](https://github.com/user-attachments/assets/90af4637-f772-4b11-843d-c9304bb55199)
Result when Inspector turns background-color off
![Header-pict-dark](https://github.com/user-attachments/assets/7c44a37f-6348-4010-95db-9551af250188)
Same for a colum
![Column-pict](https://github.com/user-attachments/assets/0fa9dc46-ff83-41ed-bcb9-75b7bfb9c34c)
And the inspector
![Column-inspector](https://github.com/user-attachments/assets/373a5a60-b892-4d11-9997-db5287f7549f)
This is how they declare the DataGrid i do not think the problem starts here but ill give it anyways.
![Declaration](https://github.com/user-attachments/assets/e7523097-b549-4927-8d72-73aebd900cad)

Tested in Chrome and Firefox with Windows 11

What could be the cause of this issue? Are we doing something wrong? Do you need more information from me? Let me know!

Thanks in advance!
Alexander

**Comment by Alexander-JM-0224** [comment]
For now i have fixed it with this.
.table-fixed-header > .table > thead:not(.table-thead-theme) > tr > th {
    background: $th-bg-color-dark !important;
}

But i still do not think this is something that is supposed to be happening, this will ultimately give problems with further styling.

**Reply by stsrki** [reply, marked as answer]
Agree. We should try using the BS5 styles as default colors for fixed headers, instead of using strongly-typed white color.

I will create a new ticket for this.

**Comment by Alexander-JM-0224** [comment]
Unfortunately this workaround is not super solid enough does interfere with other styling in our project. We look forward to a more permanent solution in the future.
